In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import re

In [2]:
trash_report = pd.read_csv('../data/trash_hauler_report.csv')

In [3]:
trash_report

,Request Number,Date Opened,Request,Description,Incident Address,Zip Code,Trash Hauler,Trash Route,Council District,State Plan X,State Plan Y
0,25270,11/1/2017,Trash - Backdoor,"house with the wheel chair ramp, they share dr...",3817 Crouch Dr,37207.0,RED RIVER,3205,2.0,1727970.412,686779.4781
1,25274,11/1/2017,Trash - Curbside/Alley Missed Pickup,Curb/Trash miss Tuesday.,4028 Clarksville Pike,37218.0,RED RIVER,4202,1.0,1721259.366,685444.7996
2,25276,11/1/2017,Trash - Curbside/Alley Missed Pickup,Curb/trash miss Tuesday.,6528 Thunderbird Dr,37209.0,RED RIVER,4205,20.0,1707026.753,659887.4716
3,25307,11/1/2017,Trash - Curbside/Alley Missed Pickup,missed,2603 old matthews rd,37207.0,WASTE IND,2206,2.0,1735691.771,685027.2459
4,25312,11/1/2017,Trash - Curbside/Alley Missed Pickup,Missed the even side of the road.,604 croley dr,37209.0,RED RIVER,4203,20.0,1710185.772,664205.1011
...,...,...,...,...,...,...,...,...,...,...,...
20221,267125,11/1/2019,Trash - Curbside/Alley Missed Pickup,MISSED...NEIGHBORS MISSED,2731 Murfreesboro Pike,37013.0,RED RIVER,4502,32.0,1781137.263,632448.5511
20222,267126,11/1/2019,Trash - Curbside/Alley Missed Pickup,entire alley,"1621 Long Ave, Nashville, TN 37206, United States",37206.0,METRO,9508,6.0,1749711.399,669201.6016
20223,267130,11/1/2019,Trash - Curbside/Alley Missed Pickup,missed several,"2943 Windemere Cir, Nashville, TN 37214, Unite...",37214.0,RED RIVER,1502,15.0,1770293.388,674936.3038
20224,267134,11/1/2019,Trash - Curbside/Alley Missed Pickup,Caller stated trash was missed & were only pic...,"3325 Murfreesboro Pike, Nashville, TN 37013, U...",37013.0,RED RIVER,4502,32.0,1785224.998,627146.4002


In [4]:
trash_report['Request '].value_counts()

Request 
Trash - Curbside/Alley Missed Pickup    15028
Trash - Backdoor                         2629
Trash Collection Complaint               2312
Damage to Property                        257
Name: count, dtype: int64

q1 
Trash - Curbside/Alley Missed Pickup    15028
Trash - Backdoor                         2629
Trash Collection Complaint               2312
Damage to Property                        257

In [5]:
trash_report['Trash Hauler'].value_counts()

Trash Hauler
RED RIVER    14395
METRO         3512
WASTE IND     1350
Metro           68
Name: count, dtype: int64

In [6]:
trash_report['Incident Address'].head()

0           3817 Crouch Dr
1    4028 Clarksville Pike
2      6528 Thunderbird Dr
3     2603 old matthews rd
4            604 croley dr
Name: Incident Address, dtype: object

In [11]:
trash_report['Incident Address'] = trash_report['Incident Address'].str.replace('Dr', 'Drive')

In [12]:
trash_report

,Request Number,Date Opened,Request,Description,Incident Address,Zip Code,Trash Hauler,Trash Route,Council District,State Plan X,State Plan Y
0,25270,11/1/2017,Trash - Backdoor,"house with the wheel chair ramp, they share dr...",3817 Crouch Drive,37207.0,RED RIVER,3205,2.0,1727970.412,686779.4781
1,25274,11/1/2017,Trash - Curbside/Alley Missed Pickup,Curb/Trash miss Tuesday.,4028 Clarksville Pike,37218.0,RED RIVER,4202,1.0,1721259.366,685444.7996
2,25276,11/1/2017,Trash - Curbside/Alley Missed Pickup,Curb/trash miss Tuesday.,6528 Thunderbird Drive,37209.0,RED RIVER,4205,20.0,1707026.753,659887.4716
3,25307,11/1/2017,Trash - Curbside/Alley Missed Pickup,missed,2603 old matthews rd,37207.0,WASTE IND,2206,2.0,1735691.771,685027.2459
4,25312,11/1/2017,Trash - Curbside/Alley Missed Pickup,Missed the even side of the road.,604 croley dr,37209.0,RED RIVER,4203,20.0,1710185.772,664205.1011
...,...,...,...,...,...,...,...,...,...,...,...
20221,267125,11/1/2019,Trash - Curbside/Alley Missed Pickup,MISSED...NEIGHBORS MISSED,2731 Murfreesboro Pike,37013.0,RED RIVER,4502,32.0,1781137.263,632448.5511
20222,267126,11/1/2019,Trash - Curbside/Alley Missed Pickup,entire alley,"1621 Long Ave, Nashville, TN 37206, United States",37206.0,METRO,9508,6.0,1749711.399,669201.6016
20223,267130,11/1/2019,Trash - Curbside/Alley Missed Pickup,missed several,"2943 Windemere Cir, Nashville, TN 37214, Unite...",37214.0,RED RIVER,1502,15.0,1770293.388,674936.3038
20224,267134,11/1/2019,Trash - Curbside/Alley Missed Pickup,Caller stated trash was missed & were only pic...,"3325 Murfreesboro Pike, Nashville, TN 37013, U...",37013.0,RED RIVER,4502,32.0,1785224.998,627146.4002


In [13]:
trash_report['Incident Address'] = trash_report['Incident Address'].str.replace('dr', 'Drive')

In [14]:
trash_report

,Request Number,Date Opened,Request,Description,Incident Address,Zip Code,Trash Hauler,Trash Route,Council District,State Plan X,State Plan Y
0,25270,11/1/2017,Trash - Backdoor,"house with the wheel chair ramp, they share dr...",3817 Crouch Drive,37207.0,RED RIVER,3205,2.0,1727970.412,686779.4781
1,25274,11/1/2017,Trash - Curbside/Alley Missed Pickup,Curb/Trash miss Tuesday.,4028 Clarksville Pike,37218.0,RED RIVER,4202,1.0,1721259.366,685444.7996
2,25276,11/1/2017,Trash - Curbside/Alley Missed Pickup,Curb/trash miss Tuesday.,6528 Thunderbird Drive,37209.0,RED RIVER,4205,20.0,1707026.753,659887.4716
3,25307,11/1/2017,Trash - Curbside/Alley Missed Pickup,missed,2603 old matthews rd,37207.0,WASTE IND,2206,2.0,1735691.771,685027.2459
4,25312,11/1/2017,Trash - Curbside/Alley Missed Pickup,Missed the even side of the road.,604 croley Drive,37209.0,RED RIVER,4203,20.0,1710185.772,664205.1011
...,...,...,...,...,...,...,...,...,...,...,...
20221,267125,11/1/2019,Trash - Curbside/Alley Missed Pickup,MISSED...NEIGHBORS MISSED,2731 Murfreesboro Pike,37013.0,RED RIVER,4502,32.0,1781137.263,632448.5511
20222,267126,11/1/2019,Trash - Curbside/Alley Missed Pickup,entire alley,"1621 Long Ave, Nashville, TN 37206, United States",37206.0,METRO,9508,6.0,1749711.399,669201.6016
20223,267130,11/1/2019,Trash - Curbside/Alley Missed Pickup,missed several,"2943 Windemere Cir, Nashville, TN 37214, Unite...",37214.0,RED RIVER,1502,15.0,1770293.388,674936.3038
20224,267134,11/1/2019,Trash - Curbside/Alley Missed Pickup,Caller stated trash was missed & were only pic...,"3325 Murfreesboro Pike, Nashville, TN 37013, U...",37013.0,RED RIVER,4502,32.0,1785224.998,627146.4002


In [16]:
trash_report['address_clean'] = trash_report['Incident Address'].str.split(',', expand=True)[0]

In [19]:
trash_report['address_clean'] = trash_report['address_clean'].str.title()

In [24]:
trash_report['address_clean'] = (trash_report['address_clean']
    .str.replace('Drive', 'Dr')
    .str.replace('Street', 'St')
    .str.replace('Road', 'Rd')
    .str.replace('Avenue', 'Ave')
    .str.replace('Boulevard', 'Blvd')
    .str.replace('Court', 'Ct')
    .str.replace('Lane', 'Ln')
    .str.replace('Place', 'Pl'))

In [26]:
trash_report['address_clean'].head(50)

0                    3817 Crouch Dr
1             4028 Clarksville Pike
2               6528 Thunderbird Dr
3              2603 Old Matthews Rd
4                     604 Croley Dr
5                    3602 Floral Dr
6                  4721 Chalmers Dr
7              1816 Jo Johnston Ave
8     4484 Lavergne Couchville Pike
9                   1206 Ireland St
10              3113 Hydes Ferry Rd
11           3210 Whites Creek Pike
12                  830 Meridian St
13                    2200 Osage St
14              6543 Thunderbird Dr
15            2218 Buena Vista Pike
16                2720 Riverbend Dr
17                 4209 Lone Oak Rd
18                  356 Kingview Dr
19                  449 Westboro Dr
20            1414 Old Hickory Blvd
21                 117 Lafayette Ct
22                1904 Meharry Blvd
23                  1144 Sewanee Rd
24               1815 Woodmont Blvd
25                      405 Page Rd
26                    4722 Danby Dr
27                  2812 27T

In [28]:
trash_report.dtypes

Request Number        int64
Date Opened          object
Request              object
Description          object
Incident Address     object
Zip Code            float64
Trash Hauler         object
Trash Route          object
Council District    float64
State Plan X        float64
State Plan Y        float64
address_clean        object
dtype: object

In [29]:
trash_report['Trash Hauler'] = trash_report['Trash Hauler'].str.upper()

In [71]:
trash_report['Date Opened'] = pd.to_datetime(trash_report['Date Opened'], format='%m/%d/%Y')

In [72]:
trash_report['Trash Hauler'].value_counts()

Trash Hauler
RED RIVER    14395
METRO         3580
WASTE IND     1350
Name: count, dtype: int64

In [73]:
trash_report['address_clean'].value_counts()

address_clean
110 George L Davis Blvd     29
5135 Hickory Hollow Pkwy    23
3710 N Natchez Ct           22
12546 Old Hickory Blvd      21
6007 Obrien Ave             20
                            ..
550 Cedar Dr                 1
4200 Pleasant Colony Dr      1
208 Mcgavock Pike            1
1500 Tremont Ave             1
604 Somerset Ct              1
Name: count, Length: 12373, dtype: int64

In [37]:
trash_report['Request '].value_counts()

Request 
Trash - Curbside/Alley Missed Pickup    15028
Trash - Backdoor                         2629
Trash Collection Complaint               2312
Damage to Property                        257
Name: count, dtype: int64

In [77]:
missed_pickups = trash_report[trash_report['Request '] == 'Trash - Curbside/Alley Missed Pickup']

In [79]:
missed_pickups_counts = missed_pickups['address_clean'].value_counts() > 1

In [80]:
missed_pickups_counts

address_clean
110 George L Davis Blvd      True
12546 Old Hickory Blvd       True
1584 Bell Rd                 True
320 Old Hickory Blvd         True
5135 Hickory Hollow Pkwy     True
                            ...  
1508 Laurel Oak Dr          False
908 Bordeaux Pl             False
572 Joyce Ln                False
2932 Hydes Ferry Rd         False
604 Somerset Ct             False
Name: count, Length: 10231, dtype: bool

In [81]:
missed_pickups_counts.sum()

np.int64(2671)

In [82]:
missed_pickups_new = missed_pickups[['address_clean', 'Trash Hauler']].value_counts().reset_index().groupby('Trash Hauler')['count'].sum()

In [83]:
missed_pickups_new.reset_index()

,Trash Hauler,count
0,METRO,2687
1,RED RIVER,10714
2,WASTE IND,1019


CONTINUE WITH SUBTRACTING THE FIRST PICK UP FROM EACH ADDRESS, CAMERON HAS THE ANSWER FOR THIS 
Q3